### PR TITLE
adding /plan endpoint to search bar

### PR DESCRIPTION
**Pull request by davidzochowski**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #[<129](https://github.com/hlxsites/danaher-ls-aem/issues/129)

Test URLs:
- Before: https://main--danaher-ls-aem--hlxsites.hlx.live/
- After: https://129-plan-coveo-api--danaher-ls-aem--hlxsites.hlx.live/

### CODE REVIEW

**Comment by davidzochowski** [comment]
Can be tested by searching for 'about us curated'. Should redirect user to the about-us page on stage.

File: blocks/header/header.js [hunk 1]
```diff
@@ -57,33 +57,39 @@ function toggleSearchBoxMobile(e) {
...
   const searchHistoryString = localStorage.getItem('__coveo.analytics.history');
   const searchHistory = searchHistoryString ? JSON.parse(searchHistoryString) : [];
   const payload = {
-    actionsHistory: searchHistory.map(({ time, value, name }) => ({ time, value, name })),
     analytics: {
       clientId,
       clientTimestamp: userTimestamp,
       documentLocation: window.location.href,
       documentReferrer: document.referrer,
       originContext: 'Search',
```
[the remark this answers is not in the document]
Comment (davidzochowski) [reply blocks/header/header.js:72]: I checked the search and the plan payloads and only the search payload has the actionsHistory object. I went with the route of having the generic payload containing what's common between them and only adding what is new for the specific ones. I added actionsHistory as part of the 'search' conditional.

[the remark this answers is not in the document]
Comment (davidzochowski) [reply blocks/header/header.js:72]: For reference, the plan api payload looks like this.
Screenshot 2023-11-06 at 9 24 28 AM
I'm ignoring capture and pageId for the time being until I can get more information about them.

File: blocks/header/header.js [hunk 2]
```diff
@@ -57,33 +57,39 @@ function toggleSearchBoxMobile(e) {
...
+
+  if (type === 'search') {
+    payload.actionsHistory = searchHistory.map(({ time, value, name }) => ({ time, value, name }));
+    payload.clientId = clientId;
+    payload.clientTimestamp = userTimestamp;
+    payload.originContext = 'Search';
+    payload.count = 8;
+    payload.referrer = document.referrer;
+  }
+  if (type === 'trigger') {
```
[the remark this answers is not in the document]
Comment (davidzochowski) [reply blocks/header/header.js:90]: Sure, I will remove it.

File: blocks/header/header.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (duynguyen) [reply blocks/header/header.js:72]: thanks David, sounds good to me :)

**Review by duynguyen — APPROVED** [review]

**Pull request merged**